src/test/java/com/google/devtools/build/lib/actions/ActionCacheCheckerTest.java

```diff
@@ -931,7 +936,8 @@ public void saveOutputMetadata_treeMetadataWithSameLocalFileMetadata_cached() th
     ActionCache.Entry entry = cache.get(output.getExecPathString());
     assertThat(entry).isNotNull();
     assertThat(entry.getOutputTree(output))
-        .isEqualTo(SerializableTreeArtifactValue.create(children, Optional.empty()));
+        .isEqualTo(
+            SerializableTreeArtifactValue.create(children, Optional.empty(), Optional.empty()));
```

Comment:
Add a test case where ActionCacheCheck is able to hit the cache for actions that produce remote tree/file symlinks?

Reply:
Done.

In passing, I've just realized that the cache checking is a little odd for artifacts with a `symlinkTargetExecPath`: we consider the local filesystem to be up to date if the artifact is materialized by an actual file/directory instead of a symlink pointing to one.

I think this is benign and consistent with how symlinks are transparently followed throughout Bazel; it's a consequence of the fact that the artifacts are declare_file/declare_directory (not declare_symlink) and hence what we really care about is their contents.